### PR TITLE
VAS migration - splitted

### PR DESCRIPTION
**Pull request by vguzmanp**
The same PR as https://github.com/fluendo/VAS/pull/3 but in two commits to avoid the GitHub limit

### CODE REVIEW

File: VAS.DB/Views/GenericView.cs [hunk 1]
```diff
@@ -20,12 +20,11 @@
...
-using LongoMatch.Core.Serialization;
-using LongoMatch.Core.Store;
 using Newtonsoft.Json.Linq;
+using VAS.Core.Common;
+using VAS.Core.Filters;
+using VAS.Core.Interfaces;
+using VAS.Core.Serialization;
 
 namespace VAS.DB.Views
 {
```
Comment (vguzmanp) [inline VAS.DB/Views/GenericView.cs]: In the part outside this diff there are references to LongoMatch

File: VAS.DB/Views/ProjectsView.cs [hunk 2]
```diff
@@ -15,12 +15,11 @@
...
 //
-using System;
 using Couchbase.Lite;
 using System.Collections.Generic;
-using LongoMatch.Core.Store;
 using Newtonsoft.Json.Linq;
 using System.Collections.Specialized;
+using VAS.Core.Store;
 
 namespace VAS.DB.Views
```
Comment (vguzmanp) [inline VAS.DB/Views/ProjectsView.cs]: VAS Projects don't have Season, Competition, ...

File: VAS.DB/Views/TimelineEventsView.cs [hunk 3]
```diff
@@ -15,13 +15,11 @@
...
 //  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
 //
-using System;
 using System.Collections;
 using System.Collections.Generic;
-using System.Linq;
 using Couchbase.Lite;
-using LongoMatch.Core.Store;
 using Newtonsoft.Json.Linq;
+using VAS.Core.Store;
```
Comment (vguzmanp) [inline VAS.DB/Views/TimelineEventsView.cs]: Are the FilterProperties needed in VAS?

File: VAS.Tests/Core/Common/TestColor.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		//		{
+		//			Color c = new Color (255, 10, 255, 10);
+		//
+		//			Utils.CheckSerialization (c);
+		//			Color c1 = Utils.SerializeDeserialize (c);
+		//			Assert.AreEqual (c, c1);
+		//
+		//			YCbCrColor yc = new YCbCrColor (2, 3, 4);
+		//			Utils.CheckSerialization (yc, true);
+		//		}
```
Comment (vguzmanp) [inline VAS.Tests/Core/Common/TestColor.cs]: commented test

File: Weavers/AssemblyInfo.cs [hunk 5]
```diff
@@ -16,23 +16,14 @@
 //  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
 //
 using System;
-using System.Collections.Generic;
-using Couchbase.Lite;
-using LongoMatch.Core.Store.Templates;
 
-namespace VAS.DB.Views
+namespace LongoMatch.Weaver
```
Comment (vguzmanp) [inline Weavers/AssemblyInfo.cs]: Shouldn't be Longomatch

File: VAS.Core/Store/Templates/Dashboard.cs [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+		const int MIN_HEIGHT = 240;
+
+		ObservableCollection<DashboardButton> list;
+
+		public Dashboard ()
+		{
+			try {
+				FieldBackground = Config.FieldBackground;
+				HalfFieldBackground = Config.HalfFieldBackground;
+				GoalBackground = Config.GoalBackground;
```
Comment (vguzmanp) [inline VAS.Core/Store/Templates/Dashboard.cs]: I don't think we should have these in VAS

File: VAS.Core/Store/Templates/Dashboard.cs [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+				Position = new Point (10 + (index % 7) * (CAT_WIDTH + 10),
+					10 + (index / 7 + 1) * (CAT_HEIGHT + 10)),
+				Width = CAT_WIDTH,
+				Height = CAT_HEIGHT,
+			};
+			List.Insert (index, button);
+			return button;
+		}
+
+		protected void FillDefaultTemplate (int count)
```
Comment (vguzmanp) [inline VAS.Core/Store/Templates/Dashboard.cs]: `DefaultTemplate` is in LongoMatch, this probably doesn't belong here

File: VAS.Core/Store/Project.cs [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+	/// <summary>
+	/// I hold the information needed by a project and provide persistency using
+	/// the db4o database.
+	/// I'm structured in the following way:
+	/// -Project Description (<see cref="LongoMatch.Utils.PreviewMediaFile"/>
+	/// -1 Categories Template
+	/// -1 Local Team Template
+	/// -1 Visitor Team Template
+	/// -1 list of <see cref="LongoMatch.Store.MediaTimeNode"/> for each category
+	/// </summary>
```
Comment (vguzmanp) [inline VAS.Core/Store/Project.cs]: Comments don't correspond to VAS

**Comment by ylatuya** [comment]
All libraries now use an AssemblyInfo.cs per library instead of a shared one like in LongoMatch

File: VAS.Core/Store/TimelineEvent.cs [hunk 9]
```diff
@@ -0,0 +1,480 @@
...
+		public Coordinates CoordinatesInFieldPosition (FieldPositionType pos)
+		{
+			switch (pos) {
+			case FieldPositionType.Field:
+				return FieldPosition;
+			case FieldPositionType.HalfField:
+				return HalfFieldPosition;
+			case FieldPositionType.Goal:
+				return GoalPosition;
+			}
```
Comment (vguzmanp) [inline VAS.Core/Store/TimelineEvent.cs]: `AddDefaultPosition` & `CoordinatesInFieldPosition` should be in LongoMatch

File: VAS.Core/Store/TimelineEvent.cs [hunk 10]
```diff
@@ -0,0 +1,480 @@
...
+				break;
+			}
+		}
+
+		// Prevents the deprecated Team field to be serialized, but allowing the field to be deserialized
+		// for migrations.
+		public bool ShouldSerializeTeam ()
+		{
+			return false;
+		}
```
Comment (vguzmanp) [inline VAS.Core/Store/TimelineEvent.cs]: There's no Team in VAS

**Pull request closed without merging**